sky/dag.py

```diff
@@ -2,74 +2,253 @@
 import pprint
 import threading
 import typing
-from typing import List, Optional
+from typing import Dict, List, Optional, Set, Union
+
+import networkx as nx
 
 if typing.TYPE_CHECKING:
     from sky import task
 
+TaskOrName = Union['task.Task', str]
+
 
 class Dag:
     """Dag: a user application, represented as a DAG of Tasks.
 
+    This class allows users to define and manage directed acyclic graphs
+    (DAGs) of tasks, representing complex workflows or pipelines.
```

Comment:
We are previously use `pipelines` as a chain of tasks. lets use `workflow` instead to reduce confusion?